gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/QueryRequestUtil.java

```diff
@@ -116,14 +117,22 @@ public static Expression createDistinctCountByColumnSelection(String columnName)
   }
 
   public static Expression createTimeColumnGroupByExpression(String timeColumn, long periodSecs) {
+    // pinot expects the time size portion to be an int. So, if it's outside the integer limit, we
```

Comment:
All of this logic really belongs in query service, since gateway shouldn't even know Pinot is the storage provider. That's obviously a bigger change though, so for now I'm curious - why hours and not just minutes? Seems like that's more than sufficient and preserves a lot more granularity

Reply:
Actually, I realized it later that this problem happens when we use Milliseconds as the time unit. But here, we're using Seconds as time unit - so we should be fine here without this change. Sorry for the confusion!